### ISSUE TITLE
Can't access microk8s via kubectl

### ISSUE THREAD

**Issue by buggtb** [ubuntu/microk8s#862]
The microk8s snap makes me sad, because I leave a perfectly fine microk8s instance running and then a few months later come back to it and something has always gone wrong this time its:

<img width="851" alt="Screenshot 2019-12-13 at 09 47 11" src="https://user-images.githubusercontent.com/103544/70820510-adb2ec80-1d8d-11ea-809c-81fbd83a4f58.png">

I can't access the pods running on it because nothing I do makes the Insufficient permissions to access MicroK8s. error go away.

**Comment by buggtb** [comment]
```
root@ased-other:~# sudo snap refresh microk8s
Consider re-refresh of "microk8s"                                                                                                                                                                                                       |
error: cannot perform the following tasks:
- Run configure hook of "microk8s" snap if present (run hook "configure":
-----
+ source /snap/microk8s/1127/actions/common/utils.sh
+ '[' '!' -d /var/snap/microk8s/1127/certs ']'
+ grep -e '\-\-client-ca-file=${SNAP}/certs/ca.crt' /var/snap/microk8s/1127/args/kube-apiserver
grep: /var/snap/microk8s/1127/args/kube-apiserver: No such file or directory
+ '[' '!' -f /var/snap/microk8s/1127/certs/csr.conf.template ']'
+ cp /snap/microk8s/1127/certs/csr.conf.template /var/snap/microk8s/1127/certs/csr.conf.template
cp: cannot create regular file '/var/snap/microk8s/1127/certs/csr.conf.template': No such file or directory
-----)
root@ased-other:~#

Then that happened... so i've thrown it all away. :(
```

**Comment by ktsakalozos** [comment]
I am sorry to hear that @buggtb . 

Would you be able to attach the tarball created by `microk8s.inspect`? Also could you share the output of `ls -l  /var/snap/microk8s/current/credentials/client.config` to see why the root user cannot access this file?

Finally let me provide some suggestions towards a more stable long lasting MicroK8s.
- Do not follow the edge channel. I see with refresh you try to get revision 1127 this means you are following the latest edge channel. Although we are doing our best to not break any channel, the edge channel gets updates on every commit we push (perhaps many times a day) so it may be unstable.
- Since you want a stable long lasting Kubernete cluster follow the stable channel of a specific upstream release, eg `sudo snap install microk8s --classic --channel=1.17/stable`. This way you will be getting patch fixes (ie 1.17.1, 1.17.2 etc) and everything else will be kept as is. By not specifying a track channel during installation you will be following the latest stable upstream release. Potential breaking changes may be introduced while we move this track from one upstream stable k8s release to another (eg from 1.15 to 1.16). These breaking changes may come from upstream (eg deprecation and removal of APIs) or from MicroK8s introducing changes in its UI. Please, have a look at [1] to read more on channels and the release process.
- Do not snap refresh to jump between channels. Snap refresh will keep the Kubernetes configuration you already have and try to apply it to the new channel. This may not always be possible.

Again apologies for breaking the deployment. We greatly value the feedback we get from early adopters following the edge channels.

[1] https://microk8s.io/docs/setting-snap-channel

**Comment by buggtb** [comment]
Sorry @ktsakalozos I removed and reinstalled it which got me a functioning install again so I don't have a dump to give you.

I wasn't actually running edge, that was just a "I'm out of ideas, and some ticket says there was a few issues in the kubectl user group stuff, so why not try it" plan, which of course failed.

It is what it is, but this same instance has fallen over irreparably twice now, when literally no one is touching it, so its a little frustrating! :)
